docker-images/kafka/scripts/kafka_config_generator.sh

```diff
@@ -6,11 +6,22 @@
 LISTENERS="REPLICATION://0.0.0.0:9091"
 ADVERTISED_LISTENERS="REPLICATION://$(hostname -f):9091"
 LISTENER_SECURITY_PROTOCOL_MAP="REPLICATION:SSL"
+SASL_ENABLED_MECHANISMS=""
 
 if [ "$KAFKA_CLIENT_ENABLED" = "TRUE" ]; then
   LISTENERS="${LISTENERS},CLIENT://0.0.0.0:9092"
   ADVERTISED_LISTENERS="${ADVERTISED_LISTENERS},CLIENT://$(hostname -f):9092"
   LISTENER_SECURITY_PROTOCOL_MAP="${LISTENER_SECURITY_PROTOCOL_MAP},CLIENT:PLAINTEXT"
+
+  if [ "$KAFKA_CLIENT_AUTHENTICATION" = "scram-sha-512" ]; then
+    SASL_ENABLED_MECHANISMS="SCRAM-SHA-512\n$SASL_ENABLED_MECHANISMS"
+    CLIENT_LISTENER=$(cat <<EOF
+# CLIENT listener authentication
+sasl.mechanism=SCRAM-SHA-512
```

Comment:
I think this sets SASL MECHANISM for outgoing client connections. Do we really need this? I thought that all we need is the `sasl.enabled.mechanisms` property. The same below for TLS. 

If it is really needed - how does Kafka handle it when this option is defined twice?

Reply:
You're right, this can be removed.